test/test_tcp_sender.py

```diff
@@ -0,0 +1,57 @@
+import queue
+from unittest import mock
+import ros_tcp_endpoint
+
+
+@mock.patch('ros_tcp_endpoint.tcp_sender.rospy')
+def test_tcp_sender_constructor(mock_ros):
+    tcp_sender = ros_tcp_endpoint.tcp_sender.UnityTcpSender()
+    assert tcp_sender.sender_id == 1
+    assert tcp_sender.time_between_halt_checks == 5
+    assert tcp_sender.queue == None
+    assert tcp_sender.next_srv_id == 1001
+    assert tcp_sender.srv_lock != None
+    assert tcp_sender.services_waiting == {}
+
+
+@mock.patch.object(ros_tcp_endpoint.tcp_sender.UnityTcpSender, 'send_unity_message')
```

Comment:
Typically it's preferable to not mock our own functions unless absolutely necessary.  In this case, I _think_ you could exercise more of our code by mocking `socket.socket` instead. This also helps better validate the contract for `send_unity_error` over the implementation details. It doesn't _have_ to call `send_unity_message`, that's an implementation detail designed to eliminate redundant code, but it _does_ have to put the error message into a socket connection, so checking for `s.send` instead creates a more targeted test (in some ways, using `socket` is also an implementation detail, but at least it's lower level)

Reply:
Make sense. I will refactor the tests.